test/unittests/validation_test.cpp

```diff
@@ -129,6 +129,170 @@ TEST(validation, memory_size_no_memory)
         parse(wasm), validation_error, "memory instructions require imported or defined memory");
 }
 
+TEST(validation, store8_alignment)
+{
+    /* wat2wasm --no-check
+    (memory 1)
+    (func (param i32)
+      get_local 0
+      i32.const 0
+      i32.store8 align=1
+    )
+    */
+    const auto wasm1 =
+        from_hex("0061736d0100000001050160017f000302010005030100010a0b010900200041003a00000b");
+    EXPECT_NO_THROW(parse(wasm1));
+
+    /* wat2wasm --no-check
+    (memory 1)
+    (func (param i32)
+      get_local 0
+      i32.const 0
+      i32.store8 align=2
+    )
+    */
+    const auto wasm2 =
+        from_hex("0061736d0100000001050160017f000302010005030100010a0b010900200041003a01000b");
+    EXPECT_THROW_MESSAGE(parse(wasm2), validation_error, "alignment cannot exceed operand size");
+
+    /* wat2wasm --no-check
+    (memory 1)
+    (func (param i32)
+      get_local 0
+      i32.const 0
+      i32.store8 align=4
+    )
+    */
+    const auto wasm3 =
+        from_hex("0061736d0100000001050160017f000302010005030100010a0b010900200041003a02000b");
+    EXPECT_THROW_MESSAGE(parse(wasm3), validation_error, "alignment cannot exceed operand size");
+}
+
+TEST(validation, store16_invalid_alignment)
+{
+    /* wat2wasm --no-check
+    (memory 1)
+    (func (param i32)
+      get_local 0
+      i32.const 0
+      i32.store16 align=1
```

Comment:
But I also mean to drop these lines to only have single failing instruction in a test case. Otherwise it is unknown which instruction the failure is coming from.

Reply:
I know, I only changed `store8` so far to check if everyone is happy before I spend 2 hrs changing the rest of the test cases.

Reply:
Ok. I suggested reducing number of cases...